test.lua

```diff
@@ -150,7 +150,7 @@ local function init_box()
     })
     a:create_index('first', {
         type = index_type,
-        parts = {1, 'NUM'},
+        parts = {1, 'unsigned'},
```

Comment:
nit: I think 'NUM' should be replaced with 'number' type. Unsigned is a subset of number.

Reply:
Also please fix count of brackets here: `parts = {1, type}` -> `parts = {{1, 'type'}}`. Or even `{{field = 1, type = type}}`.

Reply:
https://github.com/tarantool/tarantool/blob/9f208f0d3dc32ac3941090670c23d95f230ebd1a/src/box/field_def.c#L176-L191

Reply:
As I can see now on customers trying to upgrade from 1.6 (where 'num' was still allowed) to 1.10, 'num' was really an alias for 'number'.
Even though our upgrade script changes 'num' fields to 'unsigned' (which was intended. See tarantool/tarantool#1534), users may have float/double/ whatever types in former 'num' fields. This leads to errors on upgrade.

I feel like 'num' should be replaced with 'unsigned' in our system spaces only.
And for all the other spaces (user defined/testing) 'number' is a better substitution to 'num'.